lib/util/git/index.ts

```diff
@@ -250,8 +250,8 @@ export async function initRepo(args: StorageConfig): Promise<void> {
 async function resetToBranch(branchName: string): Promise<void> {
   logger.debug(`resetToBranch(${branchName})`);
   await git.raw(['reset', '--hard']);
-  await gitRetry(() => git.checkout(branchName));
-  await git.raw(['reset', '--hard', 'origin/' + branchName]);
+  await git.checkout(branchName);
```

Comment:
wrong assumption. git contacts remote if blob isn't yet downloaded. we clone with filter `blob:none` for perf

Reply:
I'd like to try reproducing that, but for now I'll just revert this piece, it doesn't make much difference anyway.

Reply:
please update PR description accordingly

Reply:
I had done this:

![image](https://github.com/user-attachments/assets/badfc205-a8ab-4d2f-b306-e8ba0d77e956)

But I removed it now.